python/ray/parameter.py

```diff
@@ -191,3 +175,8 @@ def _check_usage(self):
             assert "GPU" not in self.resources, (
                 "'GPU' should not be included in the resource dictionary. Use "
                 "num_gpus instead.")
+
+        if self.num_workers is not None:
+            raise Exception(
```

Comment:
DeprecationWarning?

Reply:
This is not new, we've been raising an exception for a long time now since 0.5 or something like that.

Reply:
OK, I see - this doesn't belong in this PR, but I think Ray can do a better job of raising more specific error messages than `Exception`